gonzo/clouds/compute.py

```diff
@@ -126,26 +126,41 @@ def create_instance(self, image_name, name, owner, user_data=None,
         if security_groups is None:
             security_groups = []
 
-        for security_group in security_groups:
-            self.create_if_not_exist_security_group(security_group)
+        for index, security_group in enumerate(security_groups):
+            self.create_if_not_exist_security_group(security_group, subnet_id)
+            security_groups[index] = self.get_security_group(
```

Comment:
this looks a bit odd. the function is called with `security_groups` but I'm not sure if these are string names or objects (maybe a docstring would help here). i'm guessing that you are replacing string names with objects, maintaining the same indexes. Is the order important? the odd bit is reassigning to the `security_groups` where I'd expect to see these objects simply appended to a new list to work with.